lib/chef/provider/package/windows/exe.rb

```diff
@@ -0,0 +1,177 @@
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# TODO: Allow @new_resource.source to be a Product Code as a GUID for uninstall / network install
+
+require 'chef/mixin/shell_out'
+
+class Chef
+  class Provider
+    class Package
+      class Windows
+        class Exe
+          include Chef::Mixin::ShellOut
+
+          def initialize(resource, installer_type)
+            @new_resource = resource
+            @installer_type = installer_type
+          end
+
+          # From Chef::Provider::Package
+          def expand_options(options)
+            options ? " #{options}" : ""
+          end
+
+          # Returns a version if the package is installed or nil if it is not.
+          def installed_version
+            Chef::Log.debug("#{@new_resource} checking package version")
+            current_installed_version
+          end
+
+          def package_version
+            @new_resource.version || install_file_version
+          end
+
+          def install_package
+            Chef::Log.debug("#{@new_resource} installing #{@new_resource.installer_type} package '#{@new_resource.source}'")
+            shell_out!(
+              [
+                "start",
+                "\"\"",
+                "/wait",
+                "\"#{@new_resource.source}\"",
+                unattended_flags,
+                expand_options(@new_resource.options),
+                "& exit %%%%ERRORLEVEL%%%%"
+              ].join(" "), timeout: @new_resource.timeout, returns: @new_resource.returns
+            )
+          end
+
+          def remove_package
+            if @new_resource.version.nil? && installed_packages.count > 1
+              raise Chef::Exceptions::MultiplePackagesFound, "Removing Windows Package '#{@new_resource.name}' found versions '#{@new_resource.version}' and no version given"
+            end
+
+            uninstall_version = @new_resource.version || installed_packages.keys[0]
+            uninstall_string = installed_packages[uninstall_version][:uninstall_string]
+
+            Chef::Log.debug("Registry provided uninstall string for #{@new_resource} is '#{uninstall_string}'")
+            shell_out!(uninstall_command(uninstall_string), { returns: @new_resource.returns })
+          end
+
+          private
+
+          def uninstall_command(uninstall_string)
+            uninstall_string.delete!('"')
+            if uninstall_string.start_with?("MsiExec.exe ")
```

Comment:
where does `MsiExec.exe ` come from? is it guaranteed to always be cased this way?